cmd/transflect-operator/operator.go

```diff
@@ -68,26 +73,51 @@ func newOperator(cfg *config) (*operator, error) {
 		k8s:   k8s,
 		istio: istio,
 
-		useIngress: cfg.UseIngress,
-		plaintext:  cfg.Plaintext,
-		address:    cfg.Address,
-		version:    "transflect-" + version,
+		useIngress:      cfg.UseIngress,
+		plaintext:       cfg.Plaintext,
+		address:         cfg.Address,
+		version:         "transflect-" + version,
+		leaderNamespace: cfg.LeaseNamespace,
 	}
 	return op, nil
 }
 
-func (o *operator) start() error {
+// start is concerned with setting up leader-election. If the current
+// process is chosen as leader, the main operator work is kicked off in
+// startLeading.
+func (o *operator) start(ctx context.Context) error {
+	// Run leader election
+	var err error
+	id := uuid.New().String()
+	ctx, cancel := context.WithCancel(ctx)
+	callbacks := leaderelection.LeaderCallbacks{
+		OnStartedLeading: func(ctx context.Context) {
+			log.Debug().Str("leaderID", id).Msg("Starting to lead")
+			if err = o.startLeading(); err != nil { // kick off operator
+				o.stop()
+				cancel()
+			}
+		},
+		OnStoppedLeading: o.stop,
```

Comment:
Having had to recently try to diagnose leader election issues, I can tell you it would be handy if this logged, particularly since you have it in the other two callbacks.

Reply:
yes.